CRM/Contribute/BAO/Contribution.php

```diff
@@ -374,7 +374,8 @@ public static function calculateMissingAmountParams(&$params, $contributionID) {
         $params['fee_amount'] = 0;
       }
     }
-    if (!isset($params['net_amount'])) {
+    // recalculate net amount if it is not set or if tax amount is set
+    if (!isset($params['net_amount']) || isset($params['tax_amount'])) {
```

Comment:
Why do we torture our poor users by asking them to enter net_amount rather than always calculating it from fee amount, tax amount etc - has anyone else felt the love of altering total & then failing to save because net amount fails validation.

Erm that was a tangent